compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/SymbolEnv.java

```diff
@@ -203,6 +203,12 @@ public static SymbolEnv createResourceActionSymbolEnv(BLangInvokableNode node, S
 
     public static SymbolEnv createArrowFunctionSymbolEnv(BLangArrowFunction node, SymbolEnv env) {
         SymbolEnv symbolEnv = cloneSymbolEnvForClosure(node, env);
+        Scope scope = node.body.scope;
+        if (scope == null) {
```

Comment:
In what sort of scenarios does this become `null`? 🤔

Reply:
This becomes `null` at the compiler level (when we creating the arrow-func-symbol environemtn